project/src/response.ts

```diff
@@ -24,6 +27,11 @@ export class Response {
     return this;
   }
 
+  withMode(mode: ResponseMode): Response {
```

Comment:
```suggestion
  withMode(mode: ResponseMode): this {
```
Hey @TomLingham, given your other PR merged, this now need to update.